From 14667a17f19c2bd769dd8303b1a67bb1fd421ea5 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Tue, 21 Sep 2021 10:49:17 +0200
Subject: [PATCH 420/446] arm64: dts: rockchip: Add various Quartz64-A board
 config variants

- Full config for SATA + WiFi + eMMC
- Full config for USB3 + WiFi + eMMC
- Bare board config for SATA

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 arch/arm64/boot/dts/rockchip/Makefile                  |  3 +++
 .../arm64/boot/dts/rockchip/rk3566-quartz64-a-base.dts | 10 ++++++++++
 .../arm64/boot/dts/rockchip/rk3566-quartz64-a-sata.dts |  7 +++++++
 .../arm64/boot/dts/rockchip/rk3566-quartz64-a-usb3.dts |  7 +++++++
 arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts     |  7 ++++---
 5 files changed, 31 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-base.dts
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-sata.dts
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-usb3.dts

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 479906f3a..00825cbcd 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -57,4 +57,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399pro-rock-pi-n10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-quartz64-a.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-quartz64-a-base.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-quartz64-a-usb3.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-quartz64-a-sata.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-evb1-v10.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-base.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-base.dts
new file mode 100644
index 000000000..a3428f3b0
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-base.dts
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+// Base DT with optional board features disabled to avoid boot log errors and slowdowns.
+// SATA is selected by default, because USB3 is broken on early Quartz64 boards.
+
+//#define USE_USB3
+//#define USE_WIFI
+//#define USE_EMMC
+
+#include "rk3566-quartz64-a.dts"
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-sata.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-sata.dts
new file mode 100644
index 000000000..e5628c40c
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-sata.dts
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+//#define USE_USB3
+#define USE_WIFI
+#define USE_EMMC
+
+#include "rk3566-quartz64-a.dts"
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-usb3.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-usb3.dts
new file mode 100644
index 000000000..c5231f643
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a-usb3.dts
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+#define USE_USB3
+#define USE_WIFI
+#define USE_EMMC
+
+#include "rk3566-quartz64-a.dts"
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
index 18a2f6bba..8176c4427 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
@@ -5,7 +5,8 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include "rk3566.dtsi"
-#define USE_SATA
+
+//#define USE_USB3
 //#define USE_WIFI
 //#define USE_EMMC
 
@@ -610,7 +611,7 @@ &pmu_io_domains {
 	vccio7-supply = <&vcc_3v3>;
 };
 
-#ifdef USE_SATA
+#ifndef USE_USB3
 /* sata1 is muxed with the usb3 port */
 &sata1 {
 	status = "okay";
@@ -750,7 +751,7 @@ &usb2phy1 {
 	status = "okay";
 };
 
-#ifndef USE_SATA
+#ifdef USE_USB3
 &usbdrd_dwc3 {
 	status = "okay";
 };
-- 
2.31.1

